frontend/src/components/custom/Question/EditQuestionCard.tsx

```diff
@@ -0,0 +1,166 @@
+import { Button } from "@/components/ui/button";
+import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { MultiSelect } from "@/components/ui/multi-select";
+import { SingleSelect } from "@/components/ui/single-select";
+import { Textarea } from "@/components/ui/textarea";
+import { SuccessObject, callFunction } from "@/lib/utils";
+import {
+  Difficulty,
+  Question,
+  QuestionExample,
+  Topic,
+  difficultyArray,
+  topicArray,
+} from "@/models/Question";
+import { Loader2 } from "lucide-react";
+import { ChangeEventHandler, FormEvent, useState } from "react";
+
+interface QuestionData {
+  id: string;
+  title: string;
+  description: string;
+  difficulty: Difficulty[];
+  topics: Topic[];
+  examples: QuestionExample[];
+  constraints: string[];
+}
+
+interface EditQuestionCardProps {
+  onEdit: () => void;
+  question: Question;
+}
+
+const EditQuestionCard: React.FC<EditQuestionCardProps> = ({
+  onEdit,
+  question,
+}) => {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [questionData, setQuestionData] = useState<QuestionData>({
+    id: question.id,
+    title: question.title,
+    description: question.description,
+    difficulty: [question.difficulty],
+    topics: question.topics,
+    examples: question.examples,
+    constraints: question.constraints,
+  });
+
+  const handleInputChange: ChangeEventHandler<HTMLInputElement> = (event) => {
+    const target = event.target as HTMLInputElement;
+    setQuestionData({
+      ...questionData,
+      [target.name]: target.value,
+    });
+  };
+
+  const handleDifficultyChange = (value: string[]) => {
+    setQuestionData({
+      ...questionData,
+      difficulty: value as Difficulty[],
+    });
+  };
+
+  const handleTopicChange = (value: string[]) => {
+    setQuestionData({
+      ...questionData,
+      topics: value as Topic[],
+    });
+  };
+
+  const handleEditQuestion = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setIsLoading(true);
+
+    const question: Partial<Question> = {
+      id: questionData.id,
+      title: questionData.title,
+      description: questionData.description,
+      topics: questionData.topics,
+      difficulty: questionData.difficulty[0],
+      examples: [],
+      constraints: [],
+    };
+
+    const result: SuccessObject = await callFunction(
+      "edit-question",
+      "POST",
```

Comment:
Can considering using `PUT` as it would be more appropriate for our use case of updating question

Reply:
noted